src/Core/src/Platform/Windows/MauiWebView.cs

```diff
@@ -143,12 +153,18 @@ void SetupPlatformEvents()
 			NavigationStarting += (sender, args) =>
 			{
 				// Auto map local virtual app dir host, e.g. if navigating back to local site from a link to an external site
-				if (args?.Uri?.ToLowerInvariant().StartsWith(LocalScheme.TrimEnd('/').ToLowerInvariant()) == true)
+				bool isUrlWithLocalScheme = args?.Uri?.ToLowerInvariant().StartsWith(LocalScheme.TrimEnd('/').ToLowerInvariant()) == true;
```

Comment:
See earlier comment about `string.Equals`

Reply:
This change just moved the if expression into a local variable. Do you want me to also change the expression, potentially introducing a bug? Or maybe do this in a separate PR?

Reply:
I added two more commits, moving the checks of local scheme URI and data URI out to static methods. See my comment above.